// react-router-addons-controlled/ControlledBrowserRouter.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ControlledHistory = require('./ControlledHistory');

var _ControlledHistory2 = _interopRequireDefault(_ControlledHistory);

var _StaticRouter = require('react-router/StaticRouter');

var _StaticRouter2 = _interopRequireDefault(_StaticRouter);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var restoreKeys = function restoreKeys() {
  try {
    return JSON.parse(sessionStorage.ReactRouterKeys);
  } catch (e) {} // eslint-disable-line
};

var saveKeys = function saveKeys(keys) {
  try {
    sessionStorage.ReactRouterKeys = JSON.stringify(keys);
  } catch (e) {} // eslint-disable-line
};

var ControlledBrowserRouter = function ControlledBrowserRouter(_ref) {
  var history = _ref.history;
  var location = _ref.location;
  var action = _ref.action;
  var onChange = _ref.onChange;
  var basename = _ref.basename;

  var routerProps = _objectWithoutProperties(_ref, ['history', 'location', 'action', 'onChange', 'basename']);

  return _react2.default.createElement(
    _ControlledHistory2.default,
    {
      history: history,
      location: location,
      action: action,
      onChange: onChange,
      restoreKeys: restoreKeys,
      saveKeys: saveKeys
    },
    function (_ref2) {
      var history = _ref2.history;
      var action = _ref2.action;
      var location = _ref2.location;
      return _react2.default.createElement(_StaticRouter2.default, _extends({
        action: action,
        location: location,
        basename: basename,
        onPush: history.push,
        onReplace: history.replace,
        blockTransitions: history.block
      }, routerProps));
    }
  );
};

ControlledBrowserRouter.propTypes = {
  history: _react.PropTypes.object,
  location: _react.PropTypes.object,
  action: _react.PropTypes.string,
  onChange: _react.PropTypes.func,
  basename: _react.PropTypes.string,
  children: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.node])
};

exports.default = ControlledBrowserRouter;
